drivers/hmis/spec/models/hmis/form/form_processor_spec.rb

```diff
@@ -36,219 +36,292 @@
     expect(assessment.enrollment.enrollment_cocs.first.coc_code).to eq('MA-507')
   end
 
-  it 'ingests IncomeBenefit into the hud tables (income sources)' do
```

Comment:
Most of the diff in this file is just changing nesting structure, and adding additional tests. Test documentation
```
Hmis::Form::FormProcessor
  ingests EnrollmentCoC into the hud tables
  pulls validation errors up from HUD records
  IncomeBenefit processor
    succeeds if overall is YES and sources are specified (income)
    succeeds if section is left empty (income)
    succeeds if overall is NO
    succeeds if overall is CLIENT_REFUSED
    succeeds if overall is YES and sources are specified (health insurance)
    succeeds if section is left empty (health insurance)
    fails if overall iS YES but no sources were specified
  HealthAndDV processor
    ingests HealthAndDV into the hud tables (no)
    ingests HealthAndDV into the hud tables (99)
    ingests HealthAndDV into the hud tables (yes, with 99 conditional)
  DisabilityGroup processor
    ingests DisabilityGroup into multiple Disabilities
    can process nil and _HIDDEN DisabilityGroup fields
  updating existing assessment
    doesn't touch an existing value, if it isn't listed (but applies the listed fields)
    clears an existing value, if it is null
    adjusts the information dates as appropriate
    adds an exit record when appropriate
    updates enrollment entry date when appropriate
  Processing PriorLivingSituation fields
    correctly sets all fields
    correctly nullifies fields
  Form processing for Clients
    creates and updates all fields
    stores empty fields correctly
    handles empty arrays
    handles Client Refused (8) and Client Doesn\t Know (9)
    fails if first and last are both nil
  Form processing for Projects
    creates and updates all fields
    stores empty fields correctly
    fails if validate if any enum field is invalid
  Form processing for Organizations
    creates and updates all fields
    stores empty fields correctly
  Form processing for Funders
    creates and updates all fields
    sets other funder
    fails if other funder not specified
    fails if end date is before start date
  Form processing for ProjectCoCs
    creates and updates all fields
  Form processing for Inventory
    creates and updates all fields
    succeeds if operating period matches project operating period
    fails if CoC code is null
    fails if CoC code is invalid for project
    fails if start date is null
    fails if end date is before start date
    fails if count is negative
    fails if start date too early
    fails if start date too early, with end date
    fails if period fully outside of project operating period
    fails if period fully outside of project operating period (other direction)
  Form processing for Service
    creates and updates all fields on HUD Service
```